handsontable/src/utils/parseTable.js

```diff
@@ -25,130 +27,551 @@ function isHTMLTable(element) {
  * @returns {string} OuterHTML of the HTMLTableElement.
  */
 export function instanceToHTML(instance) {
-  const hasColumnHeaders = instance.hasColHeaders();
-  const hasRowHeaders = instance.hasRowHeaders();
-  const coords = [
-    hasColumnHeaders ? -1 : 0,
-    hasRowHeaders ? -1 : 0,
-    instance.countRows() - 1,
-    instance.countCols() - 1,
+  const startColumn = instance.hasRowHeaders() ? -1 : 0;
+  const startRow = instance.hasColHeaders() ? -1 : 0;
+  const rows = Array.from({ length: instance.countRows() + Math.abs(startRow) },
+    (_, i) => i + startRow);
+  const columns = Array.from({ length: instance.countCols() + Math.abs(startColumn) },
+    (_, i) => i + startColumn);
+
+  return getHTMLFromHotCoords(instance, { rows, columns });
+}
+
+/**
+ * Converts Handsontable's coordinates into HTMLTableElement.
+ *
+ * @param {Core} instance The Handsontable instance.
+ * @param {object} config Configuration for building HTMLTableElement.
+ * @param {Array<number>} config.rows List of row indexes which should be taken into account when creating the table.
+ * @param {Array<number>} config.columns List of column indexes which should be taken into account when creating the table.
+ * @returns {string} OuterHTML of the HTMLTableElement.
+ */
+export function getHTMLFromHotCoords(instance, config) {
```

Comment:
```suggestion
export function getHTMLFromCoords(instance, config) {
```
In Handsontable, terms like `Coords` and `Range` are clear enough and refer to `CellCoords` and `CellRange` instances. That's why I'd remove the `Hot` postfix from the function name.